internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource.go

```diff
@@ -0,0 +1,263 @@
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return func(input interface{}, key string) (warnings []string, errors []error) {
+		v, ok := input.(string)
+		if !ok {
+			errors = append(errors, fmt.Errorf("expected %q to be a string", key))
+			return
+		}
+
+		if _, err := virtualendpoints.ParseVirtualEndpointID(v); err != nil {
+			errors = append(errors, err)
+		}
+
+		return
+	}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:        pluginsdk.TypeString,
+			Description: "The name of the Virtual Endpoint",
+			ForceNew:    true,
+			Required:    true,
+		},
+		"source_server_id": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The Resource ID of the *Source* Postgres Flexible Server this should be associated with",
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: servers.ValidateFlexibleServerID,
+		},
+		"replica_server_id": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The Resource ID of the *Replica* Postgres Flexible Server this should be associated with",
+			Required:     true,
+			ValidateFunc: servers.ValidateFlexibleServerID,
+		},
+		"type": {
+			Type:         pluginsdk.TypeString,
+			Description:  "The type of Virtual Endpoint",
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(virtualendpoints.PossibleValuesForVirtualEndpointType(), true),
+		},
+	}
+}
+
+func (r PostgresqlFlexibleServerVirtualEndpointResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var virtualEndpoint PostgresqlFlexibleServerVirtualEndpointModel
+
+			if err := metadata.Decode(&virtualEndpoint); err != nil {
+				return err
+			}
+
+			client := metadata.Client.Postgres.VirtualEndpointClient
+
+			sourceServerId, err := servers.ParseFlexibleServerID(virtualEndpoint.SourceServerId)
+			if err != nil {
+				return err
+			}
+
+			replicaServerId, err := servers.ParseFlexibleServerID(virtualEndpoint.ReplicaServerId)
+			if err != nil {
+				return err
+			}
+
+			id := virtualendpoints.NewVirtualEndpointID(sourceServerId.SubscriptionId, sourceServerId.ResourceGroupName, sourceServerId.FlexibleServerName, virtualEndpoint.Name)
+
+			locks.ByName(id.FlexibleServerName, postgresqlFlexibleServerResourceName)
+			defer locks.UnlockByName(id.FlexibleServerName, postgresqlFlexibleServerResourceName)
+
+			// This API can be a bit flaky if the same named resource is created/destroyed quickly
+			// usually waiting a minute or two before redeploying is enough to resolve the conflict
```

Comment:
Could you elaborate on this? Does the API return a conflict error or similar? We should probably raise a bug on the API specs repo if the polling isn't working properly.

Reply:
The API will return an error due to a naming collision for a few moments after deleting the previous endpoint. Waiting a few more moments will allow the system to become consistent, and then it works fine. This only occurs when deleting/recreating an endpoint with the exact same name.

I suppose a custompoller could also be used here, similar to Delete.